File: presentation/src/test/java/io/tripled/social/client/presentation/TestInput.java
package io.tripled.social.client.presentation;

import java.util.Arrays;
import java.util.Deque;
import java.util.LinkedList;

class TestInput implements Input {

  private final Deque<String> inputs;

  TestInput() {
    inputs = new LinkedList<>();
  }

  TestInput(Deque<String> inputs) {
    this.inputs = new LinkedList<>(inputs);
  }

  public void addInput(String... inputs) {
    this.inputs.addAll(Arrays.asList(inputs));
  }

  @Override
  public String read() {
    return inputs.poll();
  }
}

File: src/test/kotlin/io/tripled/social/client/infrastructure/cli/TestInput.kt
package io.tripled.social.client.infrastructure.cli

import java.util.*

internal class TestInput : Input {
    private val inputs: Deque<String?>

    init {
        inputs = LinkedList<String?>()
    }

    fun addInput(vararg inputs: String?) {
        this.inputs.addAll(Arrays.asList<String?>(*inputs))
    }

    override fun read(): String? {
        return inputs.poll()
    }
}
